Add variable support

mjablecnik opened the pull request:
<!--
  Thanks for contributing! 💙

  Provide a description of your changes below and a general summary in the title

  Please look at the following checklist to ensure that your PR can be accepted quickly:
-->

## Description

Hello I added support variables for colors and units. https://github.com/schultek/jaspr/issues/279
Right now you can define variables in your html:
```
      .raw({
        '--background-color': '#fff',
        '--text-color': Colors.blue.value,
        '--text-size': 14.pt.value,
        // other values...
      }),

```
and use it as: `Color.variable('--text-color')` or `Unit.variable('--text-size')`

## Type of Change

<!-- Uncomment all that apply: -->

<!-- - ❌ Breaking change -->
✨ New feature or improvement
<!-- - 🛠️ Bug fix -->
<!-- - 🧹 Code refactor -->
<!-- - 📝 Documentation -->
<!-- - 🗑️ Chore -->

## Ready Checklist

- [ ] I've read the [Contribution Guide](https://github.com/schultek/jaspr/blob/main/CONTRIBUTING.md).
- [x] In case this PR changes one of the core packages, I've modified the respective **CHANGELOG.md** file using
      the [semantic_changelog](https://github.com/rrousselGit/semantic_changelog) format.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added myself to the AUTHORS file (optional, if you want to).

<!-- 
  Feel free to expand this list if you have additional todos before your PR is ready. 
-->

<!--
  If you need help, consider asking for advice on the *#contribute* channel on [Discord](https://discord.gg/XGXrGEk4c6).
-->

schultek commented:
Nice change. Can you also add this for `Angle`, then it should cover probably >90% of all uses of variables.

And what do you think about calling it `.var()` instead of `.variable()`?

schultek reviewed (commented):
Nice change. Can you also add this for Angle, then it should cover probably >90% of all uses of variables.

And what do you think about calling it .var() instead of .variable()?

Dont forget to update the changelog

mjablecnik commented:
Hello, I added variable also for Angle and updated changelog.
I also wanted to call it as .var() instead of .variable() but there is problem because `var` is Dart key word:

![Screenshot 2024-09-09 at 0 09 43](https://github.com/user-attachments/assets/3336c474-a235-4ea3-ba66-8eead3383775)
vs
![Screenshot 2024-09-09 at 0 09 24](https://github.com/user-attachments/assets/2a6dc784-39e7-44a8-a6e1-0174db59be3d)
or
![Screenshot 2024-09-09 at 0 15 00](https://github.com/user-attachments/assets/16603483-da29-4fa3-89c5-db595099d222)

So we must name it somehow else then only .var()

schultek commented on packages/jaspr/CHANGELOG.md:
```diff
@@ -1,3 +1,6 @@
+## 0.16.0
```
```suggestion
## Unreleased minor
```

mjablecnik commented:
@schultek Is everything ok? Can you already merge it?

schultek reviewed (approved):
LGTM
